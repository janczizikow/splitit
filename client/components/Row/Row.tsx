import styled from "@emotion/styled";

interface Props {
  flexDirection: "row" | "column" | "row-reverse" | "column-reverse";
}

const Row = styled.div`
  margin: ${p => `0 ${p.theme.gridGutters}px`};
  padding: 0;
  display: flex;
  flex-direction: ${(p: Props) => p.flexDirection && p.flexDirection};
  flex-wrap: wrap;
  list-style: none;
`;

Row.displayName = "Row";

export default Row;
