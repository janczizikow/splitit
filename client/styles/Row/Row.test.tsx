import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Row from "./Row";

describe("<Row />", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<Row />);
    expect(wrapper).toHaveLength(1);
  });

  it("matches snapshot", () => {
    const wrapper = mount(<Row />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
