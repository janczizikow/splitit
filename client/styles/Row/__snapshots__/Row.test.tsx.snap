// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Row /> matches snapshot 1`] = `
.emotion-0 {
  margin: 0 -undefinedpx;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  list-style: none;
}

<Row>
  <div
    className="emotion-0"
  />
</Row>
`;
